src/Multipart/AbstractUploader.php

```diff
@@ -19,6 +19,8 @@
     public function __construct(Client $client, $source, array $config = [])
     {
         $this->source = $this->determineSource($source);
+        $config['exception_params']['file_name'] = $this->source->getMetadata('uri');
```

Comment:
I'd just push this responsibility off onto instances of MultipartUploadException. Have them pull it out of the `$prev` array just like how they're pulling key and bucket out.

`$prev` should always be full of UploadPart operations which, if I recall correctly, have a `Body` parameter that must be a stream. You could use that if present, pull out the metadata, and populate the filename.